mechanics/micromechanics/tests/test_agent_data.cpp

```diff
@@ -0,0 +1,165 @@
+#include <common/base_agent_data.h>
+#include <gtest/gtest.h>
+
+#include "micromechanics/agent_data.h"
+
+using namespace physicore;
+using namespace physicore::mechanics::micromechanics;
+
+class AgentDataTest : public ::testing::Test
```

Comment:
test_agent_data should verify that all arrays in agent_data have correct size after add and remove, see reactions-diffusion/biofvm/tests/test_agent_data.cpp. just have here 2 tests in the same fashion as in biofvm